### PR TITLE
fix(memory): recover observation indexing and avoid PostgreSQL pool timeouts

### PR DESCRIPTION
**Pull request by wardpeet**
Fixes cold-cache vector operations holding one PostgreSQL connection while waiting for another. Observation indexing now retries temporary failures with stable vector IDs, so a lost write acknowledgement does not create duplicate observations.

For an existing index, the same call now works with an uncached index and a single-connection pool:

```ts
const vector = new PgVector({
  id: "observations",
  connectionString,
  pgPoolOptions: { max: 1, connectionTimeoutMillis: 500 },
});

// Before: could time out waiting for a second connection.
// After: completes using the available connection.
await vector.upsert({ indexName, vectors: [[1, 0, 0]], ids: ["observation-1"] });
```

Verified with 559 passing tests (one skipped), including a real PostgreSQL regression that failed before the fix. Memory and PostgreSQL typechecks, lint, and formatting checks pass. Includes separate patch changesets.

## ELI5

This PR prevents PostgreSQL vector operations from getting stuck when the connection pool is small. It also makes observation retries update the same records instead of creating duplicates.

## Changes

- Load PostgreSQL index metadata before acquiring a connection.
- Reuse caller-owned clients during index setup without rolling back or releasing them.
- Generate stable IDs for observation embedding chunks.
- Retry observation indexing with stable vector IDs.
- Add tests for retry behavior, duplicate prevention, ID isolation, and buffering.
- Add PostgreSQL regression tests for cold-cache vector operations with one connection.
- Add patch changesets for `@mastra/memory`, `mastracode`, and `@mastra/pg`.

## Validation

- 559 tests passed.
- 1 test was skipped.
- Memory and PostgreSQL typechecks passed.
- Linting and formatting checks passed.

### CODE REVIEW

File: .changeset/full-states-move.md [hunk 1]
```diff
@@ -0,0 +1,6 @@
+---
+'@mastra/memory': patch
+'mastracode': patch
```
Comment (wardpeet) [inline .changeset/full-states-move.md]:
```suggestion
```
